### ISSUE TITLE
Cannot find element database and get_xray_lines_near_energy() 

### ISSUE THREAD

**Issue by Sl4artiB4rtF4rst** [hyperspy/exspy#45]
Dear team of Exspy! 
I really love the work you have done with exspy and hyperspy in general. I am  just getting started in evaluating and analyzing EDS (TEM) data. I wanted to use the elemental database as well as the get_xray_lines_near_energy() function but cannot find it both in the Exspy and well as the hyperspy references. I noticed proviously that some functions where relocated to other (sub) libraries - maybe herin lies the problem. 

Can you help me out where exactly to find the functions and possibly update the references as well as the EDS user guide (https://hyperspy.org/exspy/user_guide/eds.html) to show the correct usage. 

I am using hyperspy and exspy in anaconda and have installed the newest versions (from conda forge). I could not, however find the correct command to check which version I have installed exactly, sorry. 

Thank you in advance and kind regards!

**Comment by ericpre** [comment]
Here are the database:
https://github.com/hyperspy/exspy/blob/main/exspy/misc/elements.py
https://github.com/hyperspy/exspy/blob/main/exspy/misc/elements.py

And the `get_xray_lines_near_energy` function (which I found by [putting the function name in the search bar](https://github.com/search?q=repo%3Ahyperspy%2Fexspy%20get_xray_lines_near_energy&type=code) of this repository): 
https://github.com/hyperspy/exspy/blob/e70cd5cc65ffb998ffb00273fb85119cf6abf8dd/exspy/misc/eds/utils.py#L88-L124

**Comment by jlaehne** [comment]
But indeed, we should update the user guide to give working examples! Thanks for the catch.

By the way, the test datasets now reside in `exspy.data`.

**Comment by OliDG** [comment]
Hi ,

as a related problem, the "misc" module doesn't appear in the list while eXSpy is loaded and accessible. Is it a docstring issue?
 
![image](https://github.com/hyperspy/exspy/assets/35495131/daaf9a0f-ce95-4049-8040-558e825decc5)

**Comment by ericpre** [comment]
This is designed to use internally and therefore not part of the public API. Do you think that they is a use case to make it part of the public API? If so, what would it be?

**Comment by OliDG** [comment]
oooh I see, I started to code a simple procedure to get an auto-detection of the elements present in my EDS maps to simplify my work (c.f. below), 
I understand that you want to simplify the API for the public, but is there a list of all the internal API tools available somewhere? a developper documentation ?  I found this function from older hyperspy documentation but that would be a pity to re-write something already existing... 

![image](https://github.com/hyperspy/exspy/assets/35495131/1f2db201-e844-4ccf-915e-b0d3101ffcd2)
0.39  kV:  ['N_Ka', 'Sc_La', 'Ca_La', 'Ti_La']
0.52  kV:  ['O_Ka', 'V_La', 'Cr_La', 'Ti_La']
0.93  kV:  ['Cu_La', 'Ne_Kb', 'Zn_La', 'Ni_La', 'Ne_Ka']
1.49  kV:  ['Al_Ka', 'Br_La', 'Al_Kb']
8.03  kV:  ['Cu_Ka']

**Comment by OliDG** [comment]
The function 
`ex.misc.eds.utils.get_xray_lines_near_energy()`
as it is now it seems that I need to read the source code to find what is available, the dir() only shows the public side of eXSpy.
```
dir(ex)
Out[149]: 
['__version__',
 'components',
 'data',
 'material',
 'models',
 'preferences',
 'signals']
```

**Comment by ericpre** [comment]
Yes, this is expected, why should it lists what is not part of the public API?

**Comment by OliDG** [comment]
Sorry if I bother you. I am simply looking for a way to see what functions are already implemented.

**Comment by ericpre** [comment]
If you want to know more about functionalities that is _not_ public, you will need to search the source code. If you are interested in eds, it is only a few files.

**Comment by ericpre** [comment]
This should be fixed in #59: some of the functionalities were removed from the public API when splitting from hyperspy but it does make sense to add it back!
